### PR TITLE
refactor: migrate prompt .replace() chains to substitute_placeholders

### LINKED ISSUE #66: refactor: migrate prompt .replace() chains to render_template()
All prompt-rendering functions currently use chained `.replace()` calls, which risks cross-contamination when user-controlled text contains a later placeholder token.

`render_template()` in `rewrite/parsers.py` (added in #61) does single-pass regex substitution that eliminates this risk. Migrate all existing prompt renderers to use it.

**Files to update:**
- `engine/rewrite/domain_classification.py` — `_get_domain_classification_prompt`
- `engine/rewrite/sensitivity_disposition.py` — `_get_sensitivity_disposition_prompt`
- `engine/rewrite/qa_generation.py` — `_get_meaning_unit_extraction_prompt`, `_get_quality_qa_prompt`
- `engine/rewrite/rewrite_generation.py` — `_get_rewrite_prompt`
- `engine/replace/llm_replace_workflow.py` — `_get_replacement_mapping_prompt`

### PR DESCRIPTION
**Pull request by andreatnvidia**
## Summary

- Extracts `render_template` from `rewrite/parsers.py` into a shared `engine/prompt_utils.py` module, renamed to `substitute_placeholders`
- Uses single-pass regex substitution to prevent cross-contamination when user-controlled text contains placeholder tokens (e.g. `<<SCORE>>` appearing in a replacement value won't be double-substituted)
- Adds strict mode (default) that raises `ValueError` on malformed keys and unresolved placeholders, and warns on unused keys
- Migrates all 12 prompt-building functions across `detection/`, `replace/`, and `rewrite/` modules

Closes #66

## Changes

| File | Summary |
|------|---------|
| `engine/prompt_utils.py` (new) | `substitute_placeholders()` with strict/non-strict modes |
| `tests/engine/test_prompt_utils.py` (new) | 11 tests: basic substitution, cross-contamination, strict mode, non-strict mode |
| `detection/detection_workflow.py` | 3 prompt functions migrated |
| `replace/llm_replace_workflow.py` | 1 prompt function migrated |
| `rewrite/domain_classification.py` | `.replace()` chain -> `substitute_placeholders()` |
| `rewrite/evaluate.py` | import swap |
| `rewrite/final_judge.py` | import swap |
| `rewrite/parsers.py` | Removed `render_template` and unused `import re` |
| `rewrite/qa_generation.py` | 2 prompt functions migrated |
| `rewrite/repair.py` | import swap |
| `rewrite/rewrite_generation.py` | 1 prompt function migrated |
| `rewrite/sensitivity_disposition.py` | `.replace()` chain -> `substitute_placeholders()` |
| `tests/engine/test_parsers.py` | Removed `render_template` tests (moved to test_prompt_utils) |

## Test plan

- [x] All 500 existing tests pass
- [x] 11 new tests for `substitute_placeholders` covering basic substitution, cross-contamination prevention, strict mode validation, and non-strict fallback
- [x] `ruff format` and `ruff check` pass

### CODE REVIEW

File: tests/engine/test_prompt_utils.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+def test_strict_warns_on_unused_keys(caplog: pytest.LogCaptureFixture) -> None:
+    with caplog.at_level(logging.WARNING, logger="anonymizer.prompt_utils"):
+        result = substitute_placeholders("just <<A>>", {"<<A>>": "X", "<<B>>": "unused"})
+    assert result == "just X"
+    assert "<<B>>" in caplog.text
+
+
+# ---------------------------------------------------------------------------
```
Comment (lipikaramaswamy) [inline tests/engine/test_prompt_utils.py:75]: Two small additions would strengthen the contract documentation:

1. **Placeholder appearing multiple times in the template:**

```python
def test_repeated_placeholder() -> None:
    result = substitute_placeholders("<<X>> and <<X>>", {"<<X>>": "Y"})
    assert result == "Y and Y"
```

2. **Empty string as a replacement value:**

```python
def test_empty_replacement_value() -> None:
    result = substitute_placeholders("a<<X>>b", {"<<X>>": ""})
    assert result == "ab"
```

The regex approach inherently handles both, but explicit tests document the expected behavior.

File: src/anonymizer/engine/prompt_utils.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+"""Single-pass placeholder substitution for LLM prompt templates."""
+
+from __future__ import annotations
+
+import logging
+import re
+
+logger = logging.getLogger("anonymizer.prompt_utils")
+
+_PLACEHOLDER_RE = re.compile(r"<<\w+>>")
```
Comment (lipikaramaswamy) [inline src/anonymizer/engine/prompt_utils.py:14]: nit: `\w+` only matches `[a-zA-Z0-9_]`, so placeholders with hyphens (e.g. `<<DATA-CONTEXT>>`) wouldn't be recognized. All current placeholders use underscores so this is fine -- just worth a brief inline comment so future contributors know the constraint.

**Review by lipikaramaswamy — APPROVED** [review]
All looks good, workflow ran e2e, just a few nits.

File: tests/engine/test_prompt_utils.py (hunk 1 above)
Comment (andreatnvidia) [reply tests/engine/test_prompt_utils.py:75]: Added both tests - good call on documenting these edge cases explicitly.

File: src/anonymizer/engine/prompt_utils.py (hunk 2 above)
Comment (andreatnvidia) [reply src/anonymizer/engine/prompt_utils.py:14]: Good point - added an inline comment noting the constraint.
